shared/modules/types.go

```diff
@@ -1,6 +1,228 @@
-package types
+package modules
 
-// TODO(pocket/issues/163): Move this into utility
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// This file contains the minimum shared structures (GenesisState) and the many shared interfaces the modules implement
+// the main purpose of this structure is to ensure the ownership of the
+
+type GenesisState struct {
+	PersistenceGenesisState PersistenceGenesisState `json:"persistenceGenesisState"`
+	ConsensusGenesisState   ConsensusGenesisState   `json:"consensusGenesisState"`
+}
+
+type BaseConfig struct {
+	RootDirectory string `json:"root_directory"`
+	PrivateKey    string `json:"private_key"` // TODO (team) better architecture for key management (keybase, keyfiles, etc.)
+}
+
+type Config struct {
+	Base        *BaseConfig       `json:"base"`
+	Consensus   ConsensusConfig   `json:"consensus"`
+	Utility     UtilityConfig     `json:"utility"`
+	Persistence PersistenceConfig `json:"persistence"`
+	P2P         P2PConfig         `json:"p2p"`
+	Telemetry   TelemetryConfig   `json:"telemetry"`
+}
+
+type ConsensusConfig interface {
```

Comment:
Per the offline discussion around config encapsulation. What is the purpose of hiding the `struct `definition inside the module if there's going to be a 1:1 mapping between the `interface` in shared and the `struct` in the specific module?

It seems like we'll always end up needing to update two places in the code whenever we add a new field.

Reply:
If you want to swap `Olshansk's HotPocket Consensus Module` for `Andrew's TendermintBFT Consensus Module` my configuration can satisfy the interface without needing to rely on the structure in shared.

Reply:
Got it. Given how you phrased that, it makes me fear that there might be something wrong with the consensus module.

Do you see us potentially needing to revert to a TendermintBFT consensus module at some point?

Reply:
It's an example of modularity - not a comment about ownership or algorithm